File: src/CorporateMessengerWithAgile/Application/Dto/Summary/ProjectSummaryDto.cs
namespace Application.Dto.Summary
{
    public class ProjectSummaryDto : BaseDto
    {
        public Guid CompanyId { get; set; }
        public string Title { get; set; } = null!;
        public IReadOnlyList<Guid> TaskItemIds { get; set; } = null!;
        public IReadOnlyList<Guid> TeamIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/ProjectSummaryDto.ts
// src/models/entity/ProjectSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const ProjectSummaryDtoSchema = BaseDtoSchema.extend({
    companyId: GuidSchema,
    title: z.string().min(1),
    taskItemIds: z.array(GuidSchema).optional(),
    teamIds: z.array(GuidSchema).optional(),
});

export type ProjectSummaryDto = z.infer<typeof ProjectSummaryDtoSchema>;
